YAFC/Workspace/ProductionTable/ProductionTableFlatHierarchy.cs

```diff
@@ -105,6 +115,17 @@ public class FlatHierarchy<TRow, TGroup> where TRow : ModelObject<TGroup>, IGrou
             for (int i = 0; i < flatRecipes.Count; i++) {
                 var recipe = flatRecipes[i];
                 var item = flatGroups[i];
+
+                nextRowIsHighlighted = typeof(TRow) == typeof(RecipeRow) && rowHighlighting[i] != RowHighlighting.None;
```

Comment:
That needs to be expanded into serveral statements because not everyone remembers if `==` or `&&` goes first.

Reply:
That's a valid concern. Our `.editorconfig` suggests using parentheses for clarity (`dotnet_style_parentheses_in_relational_binary_operators = always_for_clarity:suggestion`).

So how about:
```csharp
nextRowIsHighlighted = (typeof(TRow) == typeof(RecipeRow)) && (rowHighlighting[i] != RowHighlighting.None);
```

Reply:
That works too.